components/sup/src/manager/user_config_watcher.rs

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::io;
+use std::path::{Path, PathBuf};
+use std::sync::mpsc::{channel, sync_channel, Sender, SendError, SyncSender, Receiver,
+                      TrySendError, TryRecvError};
+use std::thread::Builder as ThreadBuilder;
+
+use super::file_watcher::{Callbacks, default_file_watcher_with_no_initial_event};
+
+use hcore::fs::USER_CONFIG_FILE;
+use hcore::service::ServiceGroup;
+use manager::service::Service;
+use manager::service::UserConfigPath;
+
+static LOGKEY: &'static str = "UCW";
+
+// This trait exists to ease the testing of functions that receive a Service. Creating Services
+// requires a lot of ceremony, so we work around this with this trait.
+pub trait Serviceable {
+    fn name(&self) -> &str;
+    fn path(&self) -> &UserConfigPath;
```

Comment:
Might be good to call this `user_config_path` instead of just `path` for clarity.